app/controllers/inherited_proofing_controller.rb

```diff
@@ -63,6 +64,22 @@ def save_inherited_proofing_verification
     InheritedProofVerifiedUserAccount.new(user_account: @current_user.user_account).save!
   end
 
+  def validate_auth_code(auth_code)
+    raise InheritedProofing::Errors::AuthCodeMissingError unless auth_code
+
+    audit_data = inherited_proofing_audit_data(auth_code)
+    raise InheritedProofing::Errors::AuthCodeInvalidError unless audit_data
+    raise InheritedProofing::Errors::InvalidUserError unless audit_data.user_uuid == @current_user.uuid
+    raise InheritedProofing::Errors::InvalidCSPError unless
```

Comment:
I'm going back and forth on whether `legacy_csp` is necessary, especially if you're checking `user_uuid` in addition to `audit_data`. 

What I was thinking of though, if the intention is to make sure this is done with the same session that it started with, perhaps we should save the session id instead of user_uuid/legacy_csp (or oauth session uuid if we're doing SiS)?

But maybe all of this is just ideas for a future iteration. So I think for now it can probably stay as is

Reply:
Right, I was concerned about the possibility that the user could have a different session with a different CSP. I'll make a ticket to update the feature with explicit session checks.

Reply:
https://app.zenhub.com/workspaces/identity-5f5bab705a94c9001ba33734/issues/gh/department-of-veterans-affairs/va.gov-team/51137